lib/task/import/csvImportTask.class.php

```diff
@@ -439,14 +440,32 @@ public function execute($arguments = array(), $options = array())
         if ((isset($self->rowStatusVars['digitalObjectPath']) && $self->rowStatusVars['digitalObjectPath'])
           || (isset($self->rowStatusVars['digitalObjectURI']) && $self->rowStatusVars['digitalObjectURI']))
         {
-          // Delete any digital objects that exist for this information object
-          $criteria = new Criteria;
-          $criteria->add(QubitDigitalObject::INFORMATION_OBJECT_ID, $self->object->id);
-          $results = QubitDigitalObject::getOne($criteria);
+          // Retrieve any digital objects that exist for this information object
+          $do = $self->object->getDigitalObject();
 
-          if ($results !== null)
+          if ($do !== null)
           {
-            $results->delete();
+            $deleteDigitalObject = true;
+
+            if ($self->updateExisting)
+            {
+              // if   - there is a checksum in the import file
+              //      - the checksum is non-blank
+              //      - the checksum in the csv file matches what is in the database
+              // then - do not re-load the digital object from the import file on UPDATE (leave existing recs as is)
+              // else - reload the digital object in the import file (i.e. delete existing record below)
+              if ((isset($self->rowStatusVars['digitalObjectChecksum']) && $self->rowStatusVars['digitalObjectChecksum'])
+                && strcmp($self->rowStatusVars['digitalObjectChecksum'], $do->getChecksum()) === 0 )
```

Comment:
Should use yoda condition, e.g. `0 === strcmp($self->rowStatusVars['digitalObjectChecksum'], $do->getChecksum())`